### DISCUSSION TITLE
A question about how to obtian the minimum cross-entropy value.

### DISCUSSION THREAD

**Discussion by spectral0things** [lmcinnes/umap#979] [General]
If I understand correctly, the last step of UMAP minimizes a cross-entropy function between the original dataset and the resultant embedding. Is there a way to call a function to see what is the final (the minimum that UMAP achieves) value of the cross entropy that UMAP gets?

Thanks!

**Comment by jlmelville** [comment]
That is not calculated, because you only need the gradient for the optimization.

Also, be aware that the cost function being optimized is an approximation to the cross-entropy. In principle you could use the information used to calculate the gradient to get the value of this cost function (I think the related [PaCMAP](https://github.com/YingfanWang/PaCMAP) method does that as part of its implementation), but it's probably very noisy and stochastic so it would be hard to compare the final value between different runs. Maybe the parametric versions of UMAP provide a more reliable cost function?

**Reply by spectral0things** [reply]
Thanks for the response  👍
